src/extension/monitor/core/src/main/java/org/geoserver/monitor/auditlog/AuditLogger.java

```diff
@@ -80,8 +80,8 @@ public AuditLogger(MonitorConfig config, GeoServerResourceLoader loader) throws
         templateConfig.setTemplateLoader(new AuditTemplateLoader(loader));
     }
 
-    void initDumper() throws IOException {
-        if (getProperty("enabled", Boolean.class, false)) {
+    synchronized void initDumper() throws IOException {
+        if (this.dumper == null && getProperty("enabled", Boolean.class, false)) {
```

Comment:
Doh, this makes for a working double checked locking, but one that is less obvious to spot... maybe best to do the usual pattern in the caller method, "requestPostProcessed" instead?, .e.g:

```
   if (dumper == null) {
      synchronized(this) {
          if (dumper == null) initiDumper();
      }
```

Reply:
There are two places in requestPostProcessed where initDumper is called.  It seems a little messy to have to do that synchronized code twice.

Reply:
Do you have any additional thoughts about this @aaime?